src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -347,6 +366,7 @@ ucs_status_t uct_rdmacm_cm_ep_disconnect(uct_ep_h ep, unsigned flags)
         return UCS_ERR_BUSY;
     }
 
+    cep->flags |= UCT_RDMACM_CM_EP_DISCONNECTING;
```

Comment:
why is the order changed here?
what if rdma_disconnect fails?

Reply:
IMO if `rdma_disconnect` fails, no reason to call it again. We return IO_ERROR that most likely can be interpreted as not recoverable error. So, the next step of user, I would expect destroying of the EP.

Reply:
i think we should protect the setting of the cep flags since they are changed from different threads.

Reply:
done